artificial-scrum-master/apps/artificial.scrum.master.frontend/src/app/features/tasks/components/task-details/task-details.component.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  Component,
+  computed,
+  ElementRef,
+  inject,
+  Inject,
+  OnInit,
+  signal,
+  ViewChild,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { TaskDetails } from '../../models/task-details';
+import { HttpClient } from '@angular/common/http';
+import { MaterialModule } from '../../../../shared/material.module';
+import { TranslateModule, TranslateService } from '@ngx-translate/core';
+import { FormatDateService } from '../../../../shared/services/format-date.service';
+import { DescriptionDiffComponent } from '../../../../shared/components/description-diff/description-diff.component';
+import { GetStoryTaskSuggestion } from '../../models/get-story-task-suggestion';
+import { StoryTaskSuggestionService } from '../../services/story-task-suggestion.service';
+import { finalize } from 'rxjs';
+import { MarkdownEditorComponent } from '../../../../shared/components/markdown-editor/markdown-editor.component';
+import { StoryTaskEditService } from '../../services/story-task-edit.service';
+import { ToastService } from '../../../../shared/services/toast.service';
+
+@Component({
+  selector: 'app-task-details',
+  standalone: true,
+  templateUrl: './task-details.component.html',
+  styleUrls: ['./task-details.component.scss'],
+  imports: [
+    CommonModule,
+    MaterialModule,
+    TranslateModule,
+    DescriptionDiffComponent,
+    MarkdownEditorComponent,
+  ],
+})
+export class TaskDetailsComponent implements OnInit {
```

Comment:
Strasznie ogromy ten komponent. Jak mi pisałeś o 200 linijkach nie myślałem, że tak to wygląda. Teoretycznie mógłbys część logiki którą się da wynieść do serwisu typu TaskDetailsService żeby sam komponent nie musiał myśleć ale to nice to have, raczej może zostać jak jest bo nie mamy za dużo czasu

Reply:
rozdzielone na 2 komponenty